pkg/plugin/manager.go

```diff
@@ -258,9 +270,23 @@ func (m *Manager) downloadPlugin(url string) (string, error) {
 
 	downloadedFilePath := filepath.Join(tmpDir, m.downloadFilename(resp))
 
-	if err := fsutil.CopyReader(m.fs, resp.Body, downloadedFilePath, 0644); err != nil {
+	var respReader io.Reader
+	if checksum.Hasher != nil {
+		respReader = io.TeeReader(resp.Body, checksum.Hasher)
+	} else {
+		respReader = resp.Body
+	}
+	if err := fsutil.CopyReader(m.fs, respReader, downloadedFilePath, 0644); err != nil {
 		return "", err
 	}
+
+	if checksum.Hasher != nil {
+		m.logger.Debugf("Verifying checksum for %s...", url)
+		if hex.EncodeToString(checksum.Hasher.Sum(nil)) != checksum.Value {
+			return "", fmt.Errorf("computed checksum %s for %s, expected %s", hex.EncodeToString(checksum.Hasher.Sum(nil)), url, checksum.Value)
```

Comment:
This will compute a second time in case of error, it'd be a bit better to store the result in a var.